dronekit/lib/__init__.py

```diff
@@ -550,29 +526,355 @@ class Vehicle(HasObservers):
 
             ? double (0, 1, 2.3 etc...)
 
-
     .. todo:: Add waypoint_home attribute IF this is added: https://github.com/dronekit/dronekit-python/issues/105
 
     """
 
-    def __init__(self, module):
+    def __init__(self, handler):
         super(Vehicle, self).__init__()
-        self.mavrx_callback = None
-        self.__module = module
-        self._parameters = Parameters(module)
-        self._waypoints = CommandSequence(self.__module)
-        self.wpts_dirty = False
 
-    def close(self):
+        self._handler = handler
+        self._master = handler.master
+
+        # Cache all updated attributes for wait_ready.
+        # By default, we presume all "commands" are loaded.
+        self._ready_attrs = set(['commands'])
+
+        @self.attribute_listener('*')
+        def listener(_, name):
+            self._ready_attrs.add(name)
+
+        # Attaches message listeners.
+        self._message_listeners = dict()
+
+        @handler.message_listener
+        def listener(_, msg):
+            self._notify_message_listeners(msg.get_type(), msg)
+
+        self._lat = None
+        self._lon = None
+        self._vx = None
+        self._vy = None
+        self._vz = None
+
+        @self.message_listener('GLOBAL_POSITION_INT')
+        def listener(self, name, m):
```

Comment:
You should call the Ghostbusters because names attached to python functions attached to decorators are not exposed to any scope!!! Spooky :ghost: :ghost: :ghost: